src/serializer/elements.rs

```diff
@@ -528,6 +540,8 @@ impl<'a> EbmlSize for Cluster<'a> {
     }
 }
 
+// Clippy thinks this function is too complicated, but it doesn't really make sense to split it up
+#[allow(clippy::cognitive_complexity)]
```

Comment:
we might disable the cognitive complexity lint at crate-level later maybe.

Reply:
Yeah, that would make sense if more functions like the gen-functions are added.